src/Test/Perf/infra/fetch_build.csx

```diff
@@ -4,39 +4,81 @@
 #load "../util/tools_util.csx"
 
 using System;
+using System.Diagnostics;
 using System.IO;
 using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
 using System.Xml;
 
 // TODO: Use actual command line argument parser so we can have help text, etc...
-var branch = Args.Count() == 2 ? Args[0] : "master";
+var sourceFolder = Args[0];
 var destinationFolder = Args.Count() == 2 ? Args[1] : @"C:\Roslyn\Binaries\Release";
 
-var sourceFolder = $@"\\cpvsbuild\drops\Roslyn\Roslyn-{branch}-Signed-Release";
+// Get the last successful build that was tested
+string lastSuccessfulBuild = null;
+var lastSuccessfulDirectory = Environment.ExpandEnvironmentVariables($@"%SYSTEMDRIVE%\Last-Successful-{Path.GetFileName(sourceFolder)}");
+var buildInfoPath = Path.Combine((lastSuccessfulDirectory), "BuildNumber.txt");
 
-string latestBuild = null;
-foreach (var folder in Directory.GetDirectories(sourceFolder, "????????.?").Reverse())
+if (Directory.Exists(lastSuccessfulDirectory) && File.Exists(buildInfoPath))
 {
-    if (SanityTestPassesForBuild(folder))
-    {
-        latestBuild = folder;
-        break;
-    }
+    lastSuccessfulBuild = File.ReadAllText(buildInfoPath);
 }
 
-if (latestBuild == null)
+// Start monitoring the share for new successful builds
+Task.Run(() => StartWatching(sourceFolder, destinationFolder, lastSuccessfulBuild, buildInfoPath));
+Console.WriteLine("Press any key to continue");
+Console.Read();
+
+void StartWatching(
+    string sourceFolder,
+    string destinationFolder,
+    string lastSuccessfulBuild,
+    string buildInfoPath)
 {
-    throw new InvalidOperationException($"Could not locate build with passing tests at location \"{sourceFolder}\".");
-}
+    string latestBuild = null;
+    var lastSuccessfulBuildNumber = lastSuccessfulBuild == null ? 0 : Convert.ToInt32(Convert.ToDouble(lastSuccessfulBuild) * 100);
```

Comment:
Is there a reason we need to convert to ```int``` before comparing instead of just comparing ```double```?